AnkiDroid/src/main/java/com/ichi2/anki/AddonModel.java

```diff
@@ -0,0 +1,145 @@
+
+        String addonName = jsonObject.optString("name", "");
+        String addonVersion = jsonObject.optString("version", "");
+        String addonDev = jsonObject.optString("author", "");
+        String addonAnkiDroidAPI = jsonObject.optString("ankidroid_js_api", "");
+        String addonHomepage = jsonObject.optString("homepage", "");
+        String addonType = jsonObject.optString("addon_type", "");
+        if (type.equals(addonType)) {
+            return new AddonModel(addonName, addonVersion, addonDev, addonAnkiDroidAPI, addonHomepage, addonType);
+        }
+        return null;
+    }
+
+    /**
+     * @param jsonObject json object with addons info
+     * @return true/false, if valid
+     * is package.json of ankidroid-js-addon... contains valid
+     * ankidroid_js_api = 0.0.1 and keywords 'ankidroid-js-addon'
+     */
+    public static boolean isValidAddonPackage(JSONObject jsonObject, String addonType) {
+        // Update if api get updated
+        // TODO Extract to resources from other classes
+        String AnkiDroidJsAPI = "0.0.1";
+        String AnkiDroidJsAddonKeywords = "ankidroid-js-addon";
+
+        if (jsonObject == null) {
+            return false;
+        }
+
+        AddonModel addonModel = AddonModel.tryParse(jsonObject, addonType);
+        boolean jsAddonKeywordsPresent = false;
+
+        try {
+            JSONArray keywords = jsonObject.getJSONArray("keywords");
+            for (int j = 0; j < keywords.length(); j++) {
+                String addonKeyword = keywords.getString(j);
+                if (addonKeyword.equals(AnkiDroidJsAddonKeywords)) {
+                    jsAddonKeywordsPresent = true;
+                    break;
+                }
+            }
+            Timber.d("keywords %s", keywords.toString());
+        } catch (JSONException e) {
+            Timber.w(e, e.getLocalizedMessage());
+        }
+
+        if (addonModel == null) {
+            return false;
+        }
+
+        if (!addonModel.getJsApiVersion().equals(AnkiDroidJsAPI) && jsAddonKeywordsPresent) {
+            return false;
+        }
+
+        // if other strings are non empty
+        if (!addonModel.getName().isEmpty() && !addonModel.getVersion().isEmpty() && !addonModel.getDeveloper().isEmpty() && !addonModel.getHomepage().isEmpty()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public void updatePrefs(SharedPreferences preferences, String reviewerAddonKey, String addonName, boolean remove) {
+        Set<String> reviewerEnabledAddonSet = preferences.getStringSet(reviewerAddonKey, new HashSet<String>());
+        SharedPreferences.Editor editor = preferences.edit();
+
+        Set<String> newStrSet = new HashSet<String>();
+        newStrSet.addAll(reviewerEnabledAddonSet);
+        newStrSet.add(addonName);
+
+        if (remove) {
+            newStrSet.remove(addonName);
+        }
+
+        editor.putStringSet(reviewerAddonKey, newStrSet).apply();
```

Comment:
Okay, little cleaner:

```suggestion
        Set<String> newStrSet = new HashSet<String>(reviewerEnabledAddonSet);
        
        if (remove) {
            newStrSet.remove(addonName);
        } else {
            newStrSet.add(addonName);
        }

        preferences.edit().putStringSet(reviewerAddonKey, newStrSet).apply();
```